src/main/java/org/opentripplanner/routing/algorithm/raptor/router/street/AccessEgressRouter.java

```diff
@@ -0,0 +1,124 @@
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public abstract class AccessEgressRouter {
+    private static Logger LOG = LoggerFactory.getLogger(InterleavedBidirectionalHeuristic.class);
+
+    private AccessEgressRouter() {}
+
+    // TODO Switch with generic A star (abyrd)
+    public static Map<Stop, Transfer> streetSearch (RoutingRequest rr, boolean fromTarget, long abortTime) {
+        rr.maxWalkDistance = 2000;
+        rr.softWalkLimiting = false;
+        rr.softPreTransitLimiting = false;
+        Map<TransitStop, Transfer> transitStopsFound = new HashMap<>();
+        BinHeap<Vertex> transitQueue = new BinHeap<>();
+        double maxWeightSeen = 0;
+        LOG.debug("Heuristic street search around the {}.", fromTarget ? "target" : "origin");
+        rr = rr.clone();
+        if (fromTarget) {
+            rr.setArriveBy(!rr.arriveBy);
+        }
+
+        ShortestPathTree spt = new DominanceFunction.MinimumWeight().getNewShortestPathTree(rr);
+        BinHeap<State> pq = new BinHeap<State>();
+        Vertex initVertex = fromTarget ? rr.rctx.target : rr.rctx.origin;
+        State initState = new State(initVertex, rr);
+        pq.insert(initState, 0);
+        while ( ! pq.empty()) {
+            if (abortTime < Long.MAX_VALUE  && System.currentTimeMillis() > abortTime) {
+                //return null;
+            }
+            State s = pq.extract_min();
+            Vertex v = s.getVertex();
+
+            boolean initialStop = v instanceof TransitStop && s.backEdge instanceof StationStopEdge;
+
+            // At this point the vertex is closed (pulled off heap).
+            // This is the lowest cost we will ever see for this vertex. We can record the cost to reach it.
+            if (v instanceof TransitStop) {
+                transitStopsFound.put((TransitStop)v, createTransfer(s, fromTarget));
+                // We don't want to continue into the transit network yet, but when searching around the target
+                // place vertices on the transit queue so we can explore the transit network backward later.
+                if (fromTarget) {
+                    double weight = s.getWeight();
+                    transitQueue.insert(v, weight);
+                    if (weight > maxWeightSeen) {
+                        maxWeightSeen = weight;
+                    }
+                }
+                if (!initialStop) continue;
+            }
+            for (Edge e : rr.arriveBy ? v.getIncoming() : v.getOutgoing()) {
+                if (v instanceof TransitStop && !(e instanceof StreetTransitLink)) {
+                    continue;
+                }
+                // arriveBy has been set to match actual directional behavior in this subsearch.
+                // Walk cutoff will happen in the street edge traversal method.
+                State s1 = e.traverse(s);
```

Comment:
States can have next results, so it may be good to put this in a loop like so:

```java
for (State s1 = e.traverse(s); s1 != null; s1 = s1.getNextResult()) {
    // Add all states that are derived from traversing the edge.  Sometimes a fork state
    // will be encountered and those need to be added to the shortest path tree as well.
    if (spt.add(s1)) {
        pq.insert(s1,  s1.getWeight());
    }
}
```

Reply:
Tank you, just so you now, the integration code need a lot more work - but since we plan to remove all the transit code from the graph and we want something working we did this as a minimum effort.

Reply:
Just to expand a bit on this. For now the AccessEgressRouter is copied and adjusted code from the InterleavedBiDirectionalHeuristic. The plan is to replace this entirely with a generic djikstra search, as well as searches using other access/egress modes (bike, bikerental, car etc.)